Add-On UI/javascript/submit_form.html

```diff
@@ -0,0 +1,248 @@
+<!-- Javascript Functions for extracting data filled by user and 
+    calling server-side Apps Script function to get the result -->
+
+<script>
+
+  var form = document.getElementById("form");  //fetching form
+  var done = document.getElementById("done");  //fetching the submit button
+  
+  // Function called when user clicks the Compute (submit) button
+  function submitForm(intentName,event) {
+  
+    event.preventDefault();
+   
+    done.innerHTML = "Computing";   //change value of button
+    done.disabled = "true";         //disable the button
+    done.classList.add("loading");
+    
+    //intent name
+    var intent = intentName;       //intent for which query is submitted
+    
+    //table data
+    var tableData = fetchTable();  //contains table range and header row number
+    
+    //metric
+    var metric = fetchMetricData();  //contains metric column name
+
+    //summary
+    var summaryOperator = fetchSummaryData();  //summary operation applied
+    
+    //sortlimit
+    var isAsc = fetchSortOrder();  //order of sorting (only for top-k, else null)
+
+    //limit
+    var k = fetchLimit();  //number of items for top-k (else null)
+    
+    //dimension
+    var dimensions = fetchDimension();  //1D array containing name of dimensions
+    
+    //slice
+    var slices = fetchSlice();  //1D array of objects containing slicing information
+    
+    //date
+    var dateRange = fetchDate();  //date column, start date, end date
+    
+    //granularity
+    var timeGranularity = fetchGranularity();  //contains time granularity value
+    
+    //output sheet name
+    //var outputSheet = fetchOutputSheetName();  //for extracting output sheet name
```

Comment:
are we going to take output sheet name as part of the form , if yes delete the line and add a todo on yourself.

Reply:
Something like. 
// TODO(arushigarg) fetch outputname from user.

Reply:
Done